expression/builtin_string.go

```diff
@@ -1696,7 +1696,60 @@ type builtinExportSetSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.6/en/string-functions.html#function_export-set
 func (b *builtinExportSetSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("export_set")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	var (
+		bits         uint64
+		on           string
+		off          string
+		separator    = ","
+		numberOfBits = 64
+	)
+	switch len(args) {
+	case 5:
+		arg, err := args[4].ToInt64(b.ctx.GetSessionVars().StmtCtx)
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+		if arg >= 0 && arg < 64 {
+			numberOfBits = int(arg)
+		}
+		fallthrough
+	case 4:
+		separator, err = args[3].ToString()
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+		fallthrough
+	case 3:
+		arg, err := args[0].ToInt64(b.ctx.GetSessionVars().StmtCtx)
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+		bits = uint64(arg)
+		on, err = args[1].ToString()
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+		off, err = args[2].ToString()
+		if err != nil {
+			return d, errors.Trace(err)
+		}
+	}
+	res := make([]string, 0, numberOfBits)
+	for ; len(res) < numberOfBits; bits = bits / 2 {
+		var appendStr string
+		if bits&1 > 0 {
+			appendStr = on
+		} else {
+			appendStr = off
+		}
+		res = append(res, appendStr)
```

Comment:
@hanfei1991 here is my way:
```
var result string
	for i := 0; i < arg4; i++ {
		if arg0&1 == 1 {
			result = result + arg1
		} else {
			result = result + arg2
		}
		if i < arg4-1 {
			result = result + arg3
		}
		arg0 = arg0 >> 1
	}
```
Maybe more simple :laughing: 
**But there are some more problems with ToInt64() which I mentioned in  [#issue3089](https://github.com/pingcap/tidb/issues/3089)**
```
mysql> select export_set(18446744073709551616,'1','0','-');
+---------------------------------------------------------------------------------------------------------------------------------+
| export_set(18446744073709551616,'1','0','-')                                                                                  |
+---------------------------------------------------------------------------------------------------------------------------------+
| 1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1-1 |
+---------------------------------------------------------------------------------------------------------------------------------+
1 row in set, 1 warning (0.00 sec)
``` 
In MySql ,they cut the number bigger than 8 Bit to 64 bits. This would be err using `arg, err := args[0].ToInt64(b.ctx.GetSessionVars().StmtCtx) `